src/decimal.hpp

```diff
@@ -0,0 +1,64 @@
+#ifndef DECIMAL_H
+#define DECIMAL_H
+
+#include <cmath>
+#include <string>
+
+#ifdef LOST_FLOAT_MODE
+    typedef float decimal;
+    #define STR_TO_DECIMAL(x) std::stof(x)
+#else
+    typedef double decimal;
+    #define STR_TO_DECIMAL(x) std::stod(x)
+#endif
+
+// This should only be used sparingly.
+// It's better to verbosely typecast sometimes. Only use these to prevent promotions.
+// The reason why this isn't used everywhere instead of the wrapped macros is
+// because the code becomes hard to read when there are multiple layers of typecasting.
+// With this method, we might have more preprocessing to do BUT the code remains readable
+// as the methods remain relatively the same.
+#define DECIMAL(x) (decimal) x
```

Comment:
Final comment, let's make sure all these macros are "safe" (or maybe convert them to overloaded or templated functions), for example `DECIMAL(x)++` (though a very weird statement) would cause x to be incremented before being converted to decimal due to order of operations (so at minimum let's put a layer of parenthesis around the body of DECIMAL)